src/utilities.jl

```diff
@@ -238,3 +238,16 @@ function unix_joinpath(path::AbstractString, paths::AbstractString...)::String
     return path
 end
 unix_joinpath(path::AbstractString) = path
+
+"""
+    ismutable_recursive(x)
+
+Like `ismutable(x)` except it also recurses into
+`fieldnames(typeof(x))` to check if the fields are mutable.
+
+This will return `true` if `x` or any of its sub-containers
+are mutable.
+"""
+ismutable_recursive(x::T) where T = Base.ismutable(x) || any(fieldnames(T)) do field
```

Comment:
`ismutable` is 1.5+ only I _think_, hence the CI failures.

Reply:
Ah, right it should be `!isimmutable` for previous versions.